File: main/src/test/LogitBoostClassificationTest.java
package test;

import data.ClassifiedData;
import data.dataholder.DataHolder;
import data.dataholder.ObjectDataHolder;
import learning.LogitBoost;
import learning.classification.ClassificationModel;
import learning.regressors.LogisticRegressor;
import seed.SeedReader;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class LogitBoostClassificationTest {

    private static final int TRAIN_PERCENT = 80;

    public static void main(String[] args) {
        DataHolder<? extends ClassifiedData> dataHolder = new ObjectDataHolder<>(new SeedReader(), TRAIN_PERCENT, true, false);

        List<ClassifiedData> trainData = new ArrayList<>();
        List<ClassifiedData> testData = new ArrayList<>();

        List<ClassifiedData> filteredData = new ArrayList<>();
        for (ClassifiedData data : dataHolder.getData()) {
            if (data.getClassId() != 2) {
                filteredData.add(data);
            }
        }

        Collections.shuffle(filteredData);
        for (ClassifiedData data : filteredData) {
            if (trainData.size() * 100 < filteredData.size() * TRAIN_PERCENT) {
                trainData.add(data);
            } else {
                testData.add(data);
            }
        }

        LogitBoost logitBoost = new LogitBoost(() -> new LogisticRegressor(dataHolder.getVectorSize(), 0.5, 10));
        ClassificationModel classificationModel = new ClassificationModel(logitBoost, doubles -> (int) Math.round(doubles[0]));
        for (int it = 0; it < 100; it++) {
            classificationModel.trainClassified(trainData);
        }

        for (ClassifiedData data : testData) {
            System.err.printf("{%s} -> %.5f\n", data.getClassId(), classificationModel.output(data)[0]);
        }
    }
}

File: main/src/test/LogitBoostClassificationTest.kt
package test

import data.DataWithResult
import data.dataholder.ObjectDataHolder
import iris.Iris
import iris.IrisReader
import learning.LogitBoost
import learning.regressors.LogisticRegressor
import seed.SeedReader
import java.util.*

object LogitBoostClassificationTest { // TODO Rewrite it

    private val TRAIN_PERCENT = 80

    @JvmStatic fun main(args: Array<String>) {
        val dataHolder = ObjectDataHolder(SeedReader(), TRAIN_PERCENT.toDouble(), true, false)

        val trainData = ArrayList<DataWithResult>()
        val testData = ArrayList<DataWithResult>()

        val filteredData = ArrayList<DataWithResult>()
        for (data in dataHolder.data) {
            if (data.result.toInt() != 2) {
                filteredData.add(data)
            }
        }

        Collections.shuffle(filteredData)
        for (data in filteredData) {
            if (trainData.size * 100 < filteredData.size * TRAIN_PERCENT) {
                trainData.add(data)
            } else {
                testData.add(data)
            }
        }

        val logitBoost = LogitBoost { LogisticRegressor(dataHolder.vectorSize, .001, 60) }

        for (it in 0..100) {
            logitBoost.trainAll(trainData)
        }

        for (data in trainData) {
            System.err.printf("{%s} -> %.5f\n", data.result, logitBoost.output(data))
        }
    }
}
